targets/TARGET_NORDIC/TARGET_NRF5x/mbed_lib.json

```diff
@@ -1,29 +1,221 @@
 {
     "name": "nordic",
     "config": {
-        "uart-hwfc": {
-            "help": "Enable hardware flow control for STDIO.",
-            "value": 1
-        },
-        "lf-clock-src": {
-            "help": "Select Low Frequency clock source.",
+        "lf_clock_src": {
+            "macro_name": "MBED_CONF_NORDIC_NRF_LF_CLOCK_SRC",
+            "help": "Select Low Frequency clock source. Options: NRF_LF_SRC_XTAL, NRF_LF_SRC_SYNTH, and NRF_LF_SRC_RC",
             "value": "NRF_LF_SRC_XTAL"
         },
-        "uart-timeout-us": {
+        "lf_clock_rc_calib_timer_interval": {
+            "macro_name": "MBED_CONF_NORDIC_NRF_LF_CLOCK_CALIB_TIMER_INTERVAL",
+            "value": 16
+        },
+        "lf_clock_rc_calib_mode_config": {
+            "macro_name": "MBED_CONF_NORDIC_NRF_LF_CLOCK_CALIB_MODE_CONFIG",
+            "value": 0
+        },
+        "uart_hwfc": {
+            "macro_name": "MBED_CONF_NORDIC_UART_HWFC",
+            "help": "Enable hardware flow control for STDIO. Value: 1 for enable, 0 for disable",
+            "value": 0
+        },
+        "uart_timeout_us": {
             "help": "Idle time in micro seconds between characters before buffer is flushed.",
             "value": 2000
         },
-        "uart-0-fifo-size": {
+        "uart_0_fifo_size": {
             "help": "UART0 FIFO buffer. FIFO buffer is filled from DMA buffer.",
             "value": 32
         },
-        "uart-1-fifo-size": {
+        "uart_1_fifo_size": {
             "help": "UART1 FIFO buffer. FIFO buffer is filled from DMA buffer.",
             "value": 32
         },
-        "uart-dma-size": {
+        "uart_dma_size": {
             "help": "UART DMA buffer. 2 buffers per instance. DMA buffer is filled by UARTE",
             "value": 8
         }
+    },
+    "target_overrides": {
```

Comment:
Good question @loverdeg-ep

I am not certain if this change should be part of this PR. I would rather see it separately (needs also documentation where target configuration and why is it being the way it is).

Reply:
You can't override configuration from one mbed_lib.json from another mbed_lib.json. All the settings in that file are related to configuring the HAL implementations, which is why I moved it out from targets.json. 

There is a section in the README.md file about adding targets.

Reply:
Apologies,

We used mbed_app.json to overcome uart-hwfc being defined in:
```
"mbed-os/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52840/TARGET_NRF52840_DK/mbed_lib.json"
```

If this proves to be the  only solution then I'm not sure how we'll be able to make a portable and modular custom target BSP.

Will keep working at it but hoping @theotherjimmy or someone else can chime in.

Reply:
@marcuschangarm @loverdeg-ep The only way I can think of to support custom BSPs is to move the *hardware related* configuration parameters to the target.

Reply:
The thing is, that flag is a hack and not really related to the hardware.

In its original implementation, it would force all UART instances to use the RTC/CTS pins for flow control regardless whether the UART was used for STDIO or not. If you changed the interface firmware from DAPLINK to JLINK you would have to disable the flag since JLINK doesn't use flow control.

What we need is a flag in mbed_retarget.cpp to enable flow control explicitly for STDIO since that is where the STDIO instance is initialized from: https://github.com/ARMmbed/mbed-os/blob/master/platform/mbed_retarget.cpp#L148-L149

```
    serial_init(&stdio_uart, tx, rx);
    serial_baud(&stdio_uart, baud);
#if STDIO_FLOWCONTROL
    serial_set_flow_control(&stdio_uart, FlowControlRTSCTS, rts, cts);
#endif
```